Source/relay/TourGuide/Items of the Month/2022/Autumnaton.ash

```diff
@@ -71,33 +71,33 @@ void IOTMAutumnatonGenerateTasks(ChecklistEntry [int] task_entries, ChecklistEnt
 	
 	if (__misc_state["in run"] && my_path().id != 25)
 	{
-		if (locationAvailable($location[sonofa beach]) == true && available_amount($item[barrel of gunpowder]) < 5)
+		if (locationAvailable($location[sonofa beach]) == true && get_property("sidequestLighthouseCompleted") == "none" && available_amount($item[barrel of gunpowder]) < 5)
 		{
-			targets.listAppend("barrel of gunpowder");
+			targets.listAppend("barrel of gunpowder (Sonofa Beach)");
 		}
 		if (locationAvailable($location[twin peak]) == false && get_property_int("chasmBridgeProgress") < 30)
 		{
-			targets.listAppend("bridge parts");
+			targets.listAppend("bridge parts (The Smut Orc Logging Camp)");
 		}
-		if (get_property_int("hiddenBowlingAlleyProgress") < 6)
+		if (get_property_int("hiddenBowlingAlleyProgress") + available_amount($item[bowling ball]) < 6)
 		{
-			targets.listAppend("bowling balls");
+			targets.listAppend("bowling balls (The Hidden Bowling Alley)");
 		}
-		if (get_property_int("twinPeakProgress") < 14);
+		if (get_property_int("twinPeakProgress") < 14 && available_amount($item[jar of oil]) < 1 && available_amount($item[bubblin' crude]) < 12)
 		{
-			targets.listAppend("bubblin' crude");
+			targets.listAppend("bubblin' crude (Oil Peak)");
 		}
-		if (get_property_int("desertExploration") < 100);
+		if (get_property_int("desertExploration") < 100 && available_amount($item[killing jar]) < 1 && (get_property_int("gnasirProgress") & 4) == 0)
```

Comment:
what's the logic on `get_property_int("gnasirProgress") & 4`? is the & supposed to be another symbol or is there a usage of & here i don't know lol

Reply:
This one is super weird, but I copied it pretty much straight from the Mafia wiki.
https://wiki.kolmafia.us/index.php/Quest_Tracking_Preferences#gnasirProgress

Reply:
Basically the property isn't really a number, it's a set of bits where each 1 represents a completed gnasir subquest. A bitwise & will do a comparison on the bits and return the bits where they're both 1s. 

4 in binary is 0100, the quest stores completion of the killing jar subquest in the third bit. If just the killing jar subquest is complete then the property will be 4, aka 0100. 0100 & 0100 == 0100 == 4. If every subquest was complete then the property would be 31, aka 11111. 11111 & 00100 == 00100 == 4

Reply:
It's pretty clever and extremely confusing

Reply:
what in the earthly fuck lmao

Reply:
i agree you did this right btw but whoa nelly lmao. maybe we add a comment noting this and linking to that kol wiki page?